server/src/item.rs

```diff
@@ -32,8 +32,32 @@ impl Item {
     }
 
     #[zbus(out_args("secret"))]
-    pub async fn get_secret(&self, _session: ObjectPath<'_>) -> Result<SecretInner, ServiceError> {
-        todo!()
+    pub async fn get_secret(
+        &self,
+        session: OwnedObjectPath,
+    ) -> Result<(SecretInner,), ServiceError> {
+        // wip
+        let inner = self.inner.lock().await;
+        let manager = self.manager.lock().await;
+        let secret = inner.secret();
```

Comment:
Move this after getting the session, no need to "read" the secret if you will fail later anyways

Reply:
Done, thanks!